### PR TITLE
IDO MySQL: support larger host and service names

### LINKED ISSUE #7472: Support hosts with >128 characters
## Is your feature request related to a problem? Please describe.

In our cloud environment we have a challenge with long hostnames. We found out that icinga supports only hosts with a length of max. 128 characters.

Manual check execution for such hosts lead to the following error because the field name1 in icinga_objects is too short.

image

## Describe the solution you'd like

Please support hosts with hostnames >128 characters.

## Describe alternatives you've considered

I have tried to change the type of name1 in icinga_objects to varchar(256). after a restart the hosts were inserted correctly, but I reverted it at this point because we need a futureproof solution.

Tested with Icinga 2.10.5 and IcingaWeb2 2.7.1 on RHEL7.

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #7472

### CODE REVIEW

**Comment by lippserd** [comment]
Is the upgrade file missing on purpose?

**Comment by Al2Klimov** [comment]
My intention was to provide it just before the release.

**Comment by lippserd** [comment]
> My intention was to provide it just before the release.

I see. I opt to provide it right away as it makes the PR complete, test and reviewable. I know this is a door opener to merge conflicts when more PRs change the schema, but that's better than forgetting about the upgrade script or parts of it. Any objections?

**Comment by Al2Klimov** [comment]
Would be a reminder issue on v2.13.0 sufficient?

**Comment by julianbrost** [comment]
> My intention was to provide it just before the release.

So maybe now would be a good time to do this :)

**Review by lippserd — CHANGES_REQUESTED** [review]
Please provide an upgrade script.

**Comment by Al2Klimov** [comment]
@julianbrost Please could you list all v2.13 community PRs touching the schema? Preferably w/ checkboxes (whether merged).

**Comment by julianbrost** [comment]
> @julianbrost Please could you list all v2.13 community PRs touching the schema? Preferably w/ checkboxes (whether merged).

Why not handle schema migration separately for each PR that changes it? Anyways, so far I only noticed #8322 #8323 (both not merged so far)

**Comment by lippserd** [comment]
> > @julianbrost Please could you list all v2.13 community PRs touching the schema? Preferably w/ checkboxes (whether merged).
> 
> Why not handle schema migration separately for each PR that changes it? Anyways, so far I only noticed #8322 #8323 (both not merged so far)

I am totally in favor of covering schema migration in any PR that changes the schema.

**Review by lippserd — APPROVED** [review]

File: lib/db_ido_mysql/schema/upgrade/2.13.0.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+-- ----------------------------------------
+
+ALTER TABLE icinga_objects
+  MODIFY COLUMN name1 varchar(255) character set latin1 collate latin1_general_cs default '',
+  MODIFY COLUMN name2 varchar(255) character set latin1 collate latin1_general_cs default NULL;
+
+-- ------------------------------
+-- set dbversion (same as 2.11.0)
+-- ------------------------------
+INSERT INTO icinga_dbversion (name, version, create_time, modify_time) VALUES ('idoutils', '1.15.0', NOW(), NOW()) ON DUPLICATE KEY UPDATE version='1.15.0', modify_time=NOW();
```
Comment (julianbrost) [inline lib/db_ido_mysql/schema/upgrade/2.13.0.sql:23]: I'm not sure about the semantics here, but
1. shouldn't this be the same version as in `schema/mysql.sql`? (Looks like bumping in there was forgotten with the 2.11.0 schema upgrade)
2. why not make this `1.15.1`?

Comment (Al2Klimov) [reply lib/db_ido_mysql/schema/upgrade/2.13.0.sql:23]: 1. Other construction area (which we should address nevertheless).
2. Have a look at 2.8 -> 2.8.1.

Comment (julianbrost) [reply lib/db_ido_mysql/schema/upgrade/2.13.0.sql:23]: 1. -> #8852
2. Ok, then I don't get the point of this schema version but fine.

**Comment by Al2Klimov** [comment]
## Blocked by

* #8855

File: lib/db_ido_mysql/schema/upgrade/2.13.0.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+-- ----------------------------------------
+-- #7472 Support hosts with >128 characters
+-- ----------------------------------------
+
+ALTER TABLE icinga_objects
+  MODIFY COLUMN name1 varchar(255) character set latin1 collate latin1_general_cs default '',
+  MODIFY COLUMN name2 varchar(255) character set latin1 collate latin1_general_cs default NULL;
+
+-- ------------------------------
+-- set dbversion (same as 2.11.0)
```
Comment (julianbrost) [inline lib/db_ido_mysql/schema/upgrade/2.13.0.sql:21]: "same as 2.11.0" no longer applies

**Review by julianbrost — CHANGES_REQUESTED** [review]
Should also bump the schema version here: https://github.com/Icinga/icinga2/blob/b63eb2aa3dca334524185bca75ba4cd2e3ce8089/lib/db_ido_mysql/idomysqlconnection.cpp#L24-L27

**Review by julianbrost — APPROVED** [review]

**Pull request merged**